frontend/templates/product-list/server.tsx

```diff
@@ -285,11 +285,23 @@ async function getSafeServerState({
 function getDevicePathSegments(
    context: GetServerSidePropsContext<ParsedUrlQuery>
 ) {
-   const { deviceHandleItemType } = context.params || {};
-   if (Array.isArray(deviceHandleItemType)) {
-      return deviceHandleItemType;
-   }
-   return [];
+   /**
+    * We should be able to use `context.params` here,
+    * but see: https://github.com/vercel/next.js/issues/49646
+    *
+    * Basically something like device:variant%2FWithASlash
+    * Will be parsed into:
+    *   ['device', 'variant', 'WithASlash']
+    * Instead of:
+    *  ['device', 'variant/WithASlash']
+    */
+   const path = context.resolvedUrl.split('?')[0];
```

Comment:
I forget which is which, but will this be the right url when the client side is hitting the next.js `data` api to just get the props?

I'm not entirely sure we'll run into instances where that will happen for this case, but it seems feasible.

Reply:
https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props#context-parameter

>resolvedUrl
>
>A normalized version of the request URL that strips the _next/data prefix for client transitions and includes original query values.

The docs don't indicate much clarity here, but it seemed safe enough to me.

Reply:
Sounds like the right one!